crates/build/re_types_builder/src/codegen/python/mod.rs

```diff
@@ -2037,63 +2053,16 @@ fn quote_arrow_serialization(
             Ok(code)
         }
 
-        ObjectClass::Enum => {
-            // Generate case-insensitive string-to-enum conversion:
-            let match_names = obj
-                .fields
-                .iter()
-                .map(|f| {
-                    let newline = '\n';
-                    let variant = &f.name;
-                    let lowercase_variant = variant.to_lowercase();
-                    format!(
-                        r#"elif value.lower() == "{lowercase_variant}":{newline}    types.append({name}.{variant}.value)"#
-                    )
-                })
-                .format("\n")
-                .to_string();
-
-            let match_names = indent::indent_all_by(8, match_names);
-            let num_variants = obj.fields.len();
-
-            Ok(unindent(&format!(
-                r##"
+        ObjectClass::Enum => Ok(unindent(&format!(
+            r##"
 if isinstance(data, ({name}, int, str)):
     data = [data]
 
-types: list[int] = []
-
-for value in data:
-    if value is None:
-        types.append(0)
-    elif isinstance(value, {name}):
-        types.append(value.value) # Actual enum value
-    elif isinstance(value, int):
-        types.append(value) # By number
-    elif isinstance(value, str):
-        if hasattr({name}, value):
-            types.append({name}[value].value) # fast path
-{match_names}
-        else:
-            raise ValueError(f"Unknown {name} kind: {{value}}")
-    else:
-        raise ValueError(f"Unknown {name} kind: {{value}}")
-
-buffers = [
-    None,
-    pa.array(types, type=pa.int8()).buffers()[1],
-]
-children = (1 + {num_variants}) * [pa.nulls(len(data))]
+pa_data = [{name}.auto(v).value if v else None for v in data]
```

Comment:
This needs to use `if v is not None` or else ints with value 0 will convert to nulls.